src/AuthRoles.js

```diff
@@ -1,322 +1,372 @@
+          }else{
+          // new role
+            currentMapState[childRoleId] = RoleInfo(role.name, childRoleId, role.ACL, new Set([parentRoleId]));
+          }
+        });
+        // find the next ones
+        return this._findAndBuildRolesForRolesRecursivelyOntoMap(idsIterator, ids, currentMapState, masterAuth);
+      });
   }
-  return finalResult
 }
 
-// A role is accessible when any of the following statements is valid :
-// 1- User is explicitly given access to the role
-// 2- Role has access to itself
-// 3- Role is accessible from other roles we have
-// 4- Role is publicly accessible
-AuthRoles.prototype.isRoleAccessible = function(role){
-
-  // Accept role regardless of its ACL if we are using Master.
-  // Usually when using Master, we will never reach here.
-  // This is basically only used for tests.
-  if(this.isMaster === true) return true;
-
-  const acl = role.ACL;
-  const userRoles = this.accessibleRoles.names
-
-  if(acl === {} || !acl){
-    return false
-  }
-  // (1)
-  if(isAnyExplicitlyGranted(acl, [this.userId])){
-    return true
-  }
-  // (2, 4)
-  if(isAnyExplicitlyGranted(acl, ["*", "role:" + role.name])){
-    return true
-  }
-  // (3)
-  if(isAnyExplicitlyGranted(acl, userRoles)){
-    return true
-  }
-  return false
+/**
+ * A helper method to return the query to execute on _Role class
+ * @param {Object} restWhere query constraints
+ * @param {Auth} masterAuth the master auth we will be using
+ */
+const _getRolesQuery = (restWhere = {}, masterAuth: Auth) => {
+  return new RestQuery(masterAuth.config, masterAuth, '_Role', restWhere, {});
 }
 
-// Or
-function isAnyExplicitlyGranted(acl, roleNames){
-  for (let index = 0; index < roleNames.length; index++) {
-    const name = roleNames[index];
-    const statement = acl[name]
-    if(statement){
-      if(statement["read"] === true) return true
+/**
+ * Checks if ACL grants access from a Set of roles.
+ * Only one role is sufficient.
+ * @param {*} acl the acl to check
+ * @param {*} roleNames the role names to compute accessibility on 'acl'
+ * @returns {Boolean}
+ */
+const _isAclAccessibleFromRoleNames = (acl, roleNames: Set) => {
+  var isNotAccessible = true;
+  _.every(acl, (value, key) => {
+    // match name from ACL Key
+    if(roleNames.has(key)){
+      // brake when found
+      isNotAccessible = !(_isReadableAcl(value))
     }
+    return isNotAccessible;
+  })
+  return !(isNotAccessible)
+}
+
+/**
+ * Checks if ACL grants access for a specific role name.
+ * @param {*} acl the acl to check
+ * @param {*} roleName the role name to compute accessibility on 'acl'
+ * @returns {Boolean}
+ */
+const _isAclAccessibleFromRoleName = (acl, roleName) => {
+  const statement = acl[roleName];
+  if(statement){
+    return _isReadableAcl(statement);
   }
-  return false
+  return false;
 }
+
+/**
+ * Checks if acl statement is readable.
+ * "read" is true
+ * @returns {Boolean}
+ */
+const _isReadableAcl = (statement) => statement.read === true
```

Comment:
@flovilmartI don't know if this is needed. I don't recall if an ACL with r/w set to false can be available, because the read key all together will not be available. (ie "roleName" : { "write": true }. without read).
But I thought I would explicitly check the 'read' access.